### DISCUSSION TITLE
Azure blog storage integration

### DISCUSSION THREAD

**Discussion by mekya** [ant-media/Ant-Media-Server#4919] [General]

**Comment by mekya** [comment]
https://docs.microsoft.com/en-us/azure/storage/blobs/storage-how-to-mount-container-linux

**Comment by KeithLeoSmithIdeam** [comment]
Is this the intended method of integrating with azure blob storage? does it work similarly to - https://github.com/ant-media/Ant-Media-Server-Common/blob/master/src/main/java/io/antmedia/storage/AmazonS3StorageClient.java ?

**Comment by mekya** [comment]
Hi @KeithLeoSmithIdeam ,

We don't have an integration like S3(AmazonS3StorageClient.java) for Azure Blog Storage. We prefer to use fuse connectors for some Azure Blob Storage because it's simpler and more compatible

**Comment by unidentified9** [comment]
Hello @mekya, I'm trying to integrate AMS with Azure Blob Storage. I have it working with blobfuse2 by mounting it on the streams folder to get it to store m3u8 files and .ts files. However, given that AMS will delete m3u8 and .ts files, these won't be stored indefinitely in Azure Blob Storage. I'm wondering if there is a way to keep AMS deletion capabilities while also uploading these files to Azure and leaving them there for future use. 

I would prefer to do this since I don't want to use S3. Most of my stuff is on Azure already. Any thoughts?

Thanks.

**Comment by mekya** [comment]
Hi @Elang89 ,

Have you tried using the following settings in your application's red5-web.properties file?

```
settings.deleteHLSFilesOnEnded=false
```

Then restart the server to make it effective in standalone mode. 

If you use cluster, let me know there is a small trick to make this setting work. 

Regards

**Comment by unidentified9** [comment]
@mekya 

Hmm, well I was hoping that we could back up these files without having to disable that feature. Ideally we'd prefer to back things up and then let AMS remove what's not being used. 

Not sure if that's something that exists here. If there is a solution for that, I would definitely like to hear it, it would make things easier for what I'm trying to do.

**Comment by SuperElectron** [comment]
Hey I just wanted to ask a question:

- set AMS to not delete files from its system
`settings.deleteHLSFilesOnEnded=false`

- restart the server (files will still be present)

Now I assume that after restarting the server we "mirror" a copy of what we have on the server with a remote location.

Question 1: 
if we want to ensure that the AMS file system memory space is bounded (e.g Ant Media Server "System Disk" space doesn't run out of its allocated space), is there functionality with the backup to ensure that over time we keep within our expected "System disk" space?
- how can we increase system disk space?
- does back up mean that we can clear the system disk space if it grows too large without affecting our back up?

Question 2:
If we use `settings.deleteHLSFilesOnEnded=false` this means that we never delete the files off the server.  In the event where we use Ant Media Server for hundreds to thousands of streams, how can we ensure that the server memory doesn't grow extremely large?

I appreciate your time and thoughts as our client needs long term storage and we are very happy with Ant Media Server as a solution to provide video streaming with our business.

~ cheers ~

**Comment by mekya** [comment]
I'm converting this issue to a discussion to continue there.
